File: distribution-asciidoctor/src/test/java/pers/lyks/example/Swagger2markupTest.java
package pers.lyks.example;

import org.junit.Test;
import org.springframework.http.MediaType;
import org.springframework.mock.web.MockHttpServletResponse;
import org.springframework.test.web.servlet.MvcResult;

import java.io.BufferedWriter;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

/**
 * @author lawyerance
 * @version 1.0 2019-09-19
 */
public class Swagger2markupTest extends BaseRestTest {
    @Test
    public void createSpringfoxSwaggerJson() throws Exception {
        //String designFirstSwaggerLocation = Swagger2MarkupTest.class.getResource("/swagger.yaml").getPath();
        //获取生成swagger.json路径，已经在build.gradle中配置
        String outputDir = System.getProperty("io.springfox.staticdocs.outputDir");
        //本项目api路径
        MvcResult mvcResult = this.mockMvc.perform(get("/v2/api-docs")
                .accept(MediaType.APPLICATION_JSON))
                .andExpect(status().isOk())
                .andReturn();

        MockHttpServletResponse response = mvcResult.getResponse();
        String swaggerJson = response.getContentAsString();
        Files.createDirectories(Paths.get(outputDir));
        try (BufferedWriter writer = Files.newBufferedWriter(Paths.get(outputDir, "swagger.json"), StandardCharsets.UTF_8)) {
            writer.write(swaggerJson);
        }
    }
}

File: src/test/kotlin/pers/lyks/spring/example/Swagger2markupTest.kt
package pers.lyks.spring.example

import org.junit.Test
import org.springframework.http.MediaType
import org.springframework.mock.web.MockHttpServletResponse
import org.springframework.test.web.servlet.MvcResult

import java.io.BufferedWriter
import java.nio.charset.StandardCharsets
import java.nio.file.Files
import java.nio.file.Paths

import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

/**
 * @author lawyerance
 * @version 1.0 2019-09-19
 */
class Swagger2markupTest : BaseRestTest() {
    @Test
    @Throws(Exception::class)
    fun createSpringfoxSwaggerJson() {
        //String designFirstSwaggerLocation = Swagger2MarkupTest.class.getResource("/swagger.yaml").getPath();
        //获取生成swagger.json路径，已经在build.gradle中配置
        val outputDir = System.getProperty("io.springfox.staticdocs.outputDir")
        //本项目api路径
        val mvcResult = this.mockMvc.perform(get("/v2/api-docs")
                .accept(MediaType.APPLICATION_JSON))
                .andExpect(status().isOk)
                .andReturn()

        val response = mvcResult.response
        val swaggerJson = response.contentAsString
        Files.createDirectories(Paths.get(outputDir))
        Files.newBufferedWriter(Paths.get(outputDir, "swagger.json"), StandardCharsets.UTF_8).use { writer -> writer.write(swaggerJson) }
    }
}
